agent/php_api.c

```diff
@@ -1578,3 +1578,64 @@ PHP_FUNCTION(newrelic_get_trace_metadata) {
     nr_free(span_id);
   }
 }
+
+/*
+ * Purpose      Allows a caller to add a user id string to agent attributes in 
+                transaction event, transaction trace, error and span.
+ *
+ */
+#ifdef TAGS
+void zif_newrelic_set_user_id(void); /* ctags landing pad only */
+void newrelic_set_user_id(void);     /* ctags landing pad only */
+#endif
+PHP_FUNCTION(newrelic_set_user_id) {
+  zval* uuid_zv = NULL;
+  char* uuid_str = NULL;
+
+  NR_UNUSED_RETURN_VALUE_PTR;
+  NR_UNUSED_THIS_PTR;
+  NR_UNUSED_RETURN_VALUE_USED;
+
```

Comment:
We should probably bail out early if we aren't recording, like this: https://github.com/newrelic/newrelic-php-agent/blob/dev/agent/php_api.c#L650-L652

Reply:
Wouldn't that case already be handled by checking that the transaction is not NULL on line [1601](https://github.com/newrelic/newrelic-php-agent/pull/723/files#diff-aaa6dfffdf27a5da3f0c89176d8aa9b62226676fae89c5c2c8e0e1a35db06b66R1601)?

Reply:
Or is it possible to create a transaction object but not record anything?

Reply:
Ah I suppose someone could hypothetically place a call to `nr_txn_ignore()` as well as a call to this API (although I'm not sure why somebody would do that) so checking to see if the transaction is recording is technically necessary for cases like that.

Reply:
Added in e8f8ac25d51924c2173e529db729f387ed5353e5